src/drivers/display/nxp_elcdif/lv_nxp_elcdif.c

```diff
@@ -0,0 +1,185 @@
+}
+
+lv_display_t * lv_nxp_display_elcdif_create_partial(LCDIF_Type * base, const elcdif_rgb_mode_config_t * config,
+                                                    void * frame_buffer1,
+                                                    void * frame_buffer2, size_t buf_size)
+{
+    LV_ASSERT(base);
+    LV_ASSERT(config);
+
+    /* Create a direct mode display and then update the buffers to be set in partial mode */
+    lv_display_t * disp = lv_nxp_display_elcdif_create_direct(base, config, frame_buffer1, frame_buffer2, buf_size);
+    ELCDIF_DisableInterrupts(base, kELCDIF_CurFrameDoneInterruptEnable);
+    NVIC_DisableIRQ(eLCDIF_IRQn);
+
+    lv_display_set_flush_cb(disp, flush_partial_cb);
+    lv_display_set_buffers(disp, frame_buffer1, frame_buffer2, buf_size, LV_DISPLAY_RENDER_MODE_PARTIAL);
+
+    ELCDIF_EnableInterrupts(base, kELCDIF_CurFrameDoneInterruptEnable);
+    NVIC_EnableIRQ(eLCDIF_IRQn);
+
+    return disp;
+}
+
+void lv_nxp_display_elcdif_event_handler(const lv_display_t * disp)
+{
+    if(disp == NULL) {
+        /* Just return since no valid display has been set yet */
+        return;
+    }
+
+    LCDIF_Type * base = (LCDIF_Type *)lv_display_get_user_data((lv_display_t *)disp);
+    uint32_t intStatus = ELCDIF_GetInterruptStatus(base);
+
+    ELCDIF_ClearInterruptStatus(base, intStatus);
+
+    if(intStatus & kELCDIF_CurFrameDone) {
+        /* flush ready is ISR safe and atomic, so calling inside of the
+         * framebuffer interrupt is safe and makes the flush chain
+         * non blocking even in bare metal systems.
+         */
+        lv_disp_flush_ready((lv_display_t *)disp);
+    }
+
+    SDK_ISR_EXIT_BARRIER;
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * color_p)
+{
+    LCDIF_Type * base = (LCDIF_Type *)lv_display_get_user_data(disp);
+
+    DCACHE_CleanInvalidateByRange((uint32_t)color_p, lv_display_get_draw_buf_size(disp));
+
+    if(!lv_display_flush_is_last(disp)) {
+        lv_disp_flush_ready(disp);
+        return;
+    }
+    ELCDIF_SetNextBufferAddr(base, (uint32_t)color_p);
+}
+
+static void flush_partial_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * color_p)
+{
+    LCDIF_Type * base = (LCDIF_Type *)lv_display_get_user_data(disp);
+
+    DCACHE_CleanInvalidateByRange((uint32_t)color_p, lv_display_get_draw_buf_size(disp));
+
+    uint8_t * fb = (uint8_t *)ELCDIF_ADDR_IP_2_CPU(base->CUR_BUF);
+    int32_t w = lv_area_get_width(area);
+    int32_t h = lv_area_get_height(area);
+    int32_t disp_w = lv_display_get_horizontal_resolution(disp);
+    int32_t disp_h = lv_display_get_vertical_resolution(disp);
+    int32_t bytes_per_pixel = LV_COLOR_FORMAT_GET_SIZE(lv_display_get_color_format(disp));
+    int32_t i;
+
+    fb = fb + area->y1 * disp_h;
+    fb = fb + area->x1;
+
+    for(i = 0; i < h; i++) {
+        lv_memcpy(fb, color_p, w * bytes_per_pixel);
```

Comment:
It doesn't make use of 2 buffers this way, but let's go for it for now. We will see how to improve it later. Maybe with DMA.